Src/CSharpier.Cli/Options/ConfigurationFileOptions.cs

```diff
@@ -16,128 +12,70 @@ internal class ConfigurationFileOptions
+    public Override[] Overrides { get; init; } = [];
 
-    internal static PrinterOptions CreatePrinterOptionsFromPath(
-        string configPath,
-        IFileSystem fileSystem,
-        ILogger logger
-    )
+    public PrinterOptions? ConvertToPrinterOptions(string filePath)
     {
-        var configurationFileOptions = Create(configPath, fileSystem, logger);
-
-        return ConvertToPrinterOptions(configurationFileOptions);
-    }
-
-    internal static PrinterOptions ConvertToPrinterOptions(
-        ConfigurationFileOptions configurationFileOptions
-    )
-    {
-        return new PrinterOptions
-        {
-            TabWidth = configurationFileOptions.TabWidth,
-            UseTabs = configurationFileOptions.UseTabs,
-            Width = configurationFileOptions.PrintWidth,
-            EndOfLine = configurationFileOptions.EndOfLine
-        };
-    }
-
-    /// <summary>Finds all configs above the given directory as well as within the subtree of this directory</summary>
-    internal static List<CSharpierConfigData> FindForDirectoryName(
-        string directoryName,
-        IFileSystem fileSystem,
-        ILogger logger,
-        bool limitEditorConfigSearch
-    )
-    {
-        var results = new List<CSharpierConfigData>();
-        var directoryInfo = fileSystem.DirectoryInfo.New(directoryName);
-
-        var filesByDirectory = directoryInfo
-            .EnumerateFiles(
-                ".csharpierrc*",
-                limitEditorConfigSearch
-                    ? SearchOption.TopDirectoryOnly
-                    : SearchOption.AllDirectories
-            )
-            .GroupBy(o => o.DirectoryName);
-
-        foreach (var group in filesByDirectory)
+        DebugLogger.Log("finding options for " + filePath);
+        var matchingOverride = this.Overrides.LastOrDefault(o => o.IsMatch(filePath));
+        if (matchingOverride is not null)
         {
-            var firstFile = group
-                .Where(o => validExtensions.Contains(o.Extension, StringComparer.OrdinalIgnoreCase))
-                .MinBy(o => o.Extension);
-
-            if (firstFile != null)
+            return new PrinterOptions
             {
-                results.Add(
-                    new CSharpierConfigData(
-                        firstFile.DirectoryName!,
-                        Create(firstFile.FullName, fileSystem, logger)
-                    )
-                );
-            }
+                TabWidth = matchingOverride.TabWidth,
+                UseTabs = matchingOverride.UseTabs,
+                Width = matchingOverride.PrintWidth,
+                EndOfLine = matchingOverride.EndOfLine,
+                Formatter = matchingOverride.Formatter
+            };
         }
 
-        // already found any in this directory above
-        directoryInfo = directoryInfo.Parent;
-
-        while (directoryInfo is not null)
+        if (filePath.EndsWith(".cs") || filePath.EndsWith(".csx"))
```

Comment:
this check has come up in multiple places (including `EditorConfigSections.cs`). Do you want to extract the extensions to a constant?
Or even create a helper method?

Reply:
I need to do something with it in https://github.com/belav/csharpier/pull/858/files, because that is going to add csproj/xml formatting. Maybe that is the time to switch the CLI option to use globs. I'll leave it for now and deal with it there, I'm sure there are plenty of conflicts on those lines already.